engine/execution/pruner/core.go

```diff
@@ -155,3 +164,18 @@ func makeBlockIteratorCreator(
 		return next, header.Height, nil
 	}, nil
 }
+
+// estimateBatchProcessing estimates the number of batches and the total duration
+func EstimateBatchProcessing(
+	start, end uint64, batchSize uint, sleepAfterEachBatchCommit time.Duration, commitDuration time.Duration) (
+	batchCount uint64, totalDuration time.Duration) {
+	if batchSize == 0 || start >= end {
+		return 0, 0
+	}
+
+	batchCount = (end - start + uint64(batchSize) - 1) / uint64(batchSize)
```

Comment:
Maybe adjust remaining count to `end - start + 1` because `end` is inclusive.

```suggestion
	count := end - start + 1
	batchCount = (count + uint64(batchSize) - 1) / uint64(batchSize)
```